internal/app/storage/models/metadata.go

```diff
@@ -0,0 +1,34 @@
+package models
```

Comment:
Mmm, this kind of package name is considered as a bad practice.
Why not let models at "pkg" root (or "internal")? 
From a dev perspective, it is easy to quickly see models of the application.